openshift/machine-config-operator
# MCO-1231: MCO-783: MCO-1232: Use Jobs for OCL builds

**- What I did**
Instead of using just a Pod type for the image builds with OCL, Use Jobs. This allows us to use the job controller to our advantage when it comes to retries on failures, cleanups, pod evictions due to constrained resources etc.
The build controller creates a Job when a new build is triggered. The Job creates a pod to run the builds and watches its status to take appropriate action based on success and failure. Once the build is successful, the job and its pods are cleaned up. If the build fails, the job and any pods it created stay around to allow the user the opportunity to debug. The Job will retry the build 4 times before reporting as Failed. Each time it fails, a new pod will be created for the retry process.
Add appropriate unit and e2e tests.

**- How to verify it**
Enable the cluster to use OCL builds and create an MOSC to trigger a build. A Job should be created for the build process.

**- Description for the changelog**
Use Kubernetes Jobs instead of just a Pod for the build process during OCL.
